test/Core.Test/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommandTests.cs

```diff
@@ -0,0 +1,631 @@
+            .GetByIdAsync(organizationId)
+            .Returns(organization);
+
+        var organizationUpdate = new OrganizationUpdate
+        {
+            OrganizationId = organization.Id,
+            MaxAutoscaleSeats = 15,
+            SeatAdjustment = 1,
+            MaxAutoscaleServiceAccounts = 300,
+            ServiceAccountsAdjustment = 1
+        };
+
+        sutProvider.GetDependency<IOrganizationRepository>()
+            .GetByIdAsync(organization.Id)
+            .Returns(organization);
+
+        var plan = StaticStore.SecretManagerPlans.Single(x => x.Type == organization.PlanType);
+        plan.HasAdditionalServiceAccountOption = false;
+        var exception = await Assert.ThrowsAsync<BadRequestException>(() => sutProvider.Sut.UpdateSecretsManagerSubscription(organizationUpdate));
+        Assert.Contains("Plan does not allow additional Service Account.", exception.Message, StringComparison.InvariantCultureIgnoreCase);
+    }
+
+
+    [Theory]
+    [BitAutoData(PlanType.TeamsAnnually)]
+    [BitAutoData(PlanType.TeamsMonthly)]
+    [BitAutoData(PlanType.EnterpriseAnnually)]
+    [BitAutoData(PlanType.EnterpriseMonthly)]
+    public async Task UpdateSecretsManagerSubscription_ValidInput_Passes(
+        PlanType planType,
+        Guid organizationId,
+        SutProvider<UpdateSecretsManagerSubscriptionCommand> sutProvider)
+    {
+        var organization = new Organization
+        {
+            Id = organizationId,
+            SmSeats = 10,
+            SmServiceAccounts = 200,
+            MaxAutoscaleSmSeats = 20,
+            MaxAutoscaleSmServiceAccounts = 300,
+            PlanType = planType,
+            GatewayCustomerId = "1",
+            GatewaySubscriptionId = "2"
+        };
+
+        var previousSeats = organization.SmSeats;
+        var previousServiceAccounts = organization.SmServiceAccounts;
+        sutProvider.GetDependency<IOrganizationRepository>()
+            .GetByIdAsync(organizationId)
+            .Returns(organization);
+
+        var organizationUpdate = new OrganizationUpdate
+        {
+            OrganizationId = organizationId,
+            MaxAutoscaleSeats = 15,
+            SeatAdjustment = 5,
+            MaxAutoscaleServiceAccounts = 300,
+            ServiceAccountsAdjustment = 100
+        };
+        var plan = StaticStore.SecretManagerPlans.FirstOrDefault(x => x.Type == organization.PlanType);
+        plan.HasAdditionalSeatsOption = true;
+        plan.HasAdditionalServiceAccountOption = true;
+        var additionalSeats = (organization.SmSeats + organizationUpdate.SeatAdjustment) - plan.BaseSeats;
+        var additionalServiceAccounts =
+            (organization.SmServiceAccounts + organizationUpdate.ServiceAccountsAdjustment) - plan.BaseServiceAccount;
+        var newSeatTotal = organization.SmSeats.HasValue
+            ? organization.SmSeats.Value + organizationUpdate.SeatAdjustment
+            : organizationUpdate.SeatAdjustment;
```

Comment:
Use `GetValueOrDefault()` to simplify null handling throughout these tests

Reply:
resolved